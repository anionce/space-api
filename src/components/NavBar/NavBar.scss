@import '../../assets/styles/variables';

nav {
	display: flex;
	flex-direction: row;
	justify-content: center;
	gap: 16px;
	margin-bottom: 36px;

	a {
		color: $color-primary;
		font-weight: 600;
		text-decoration: none;
		text-transform: uppercase;
		font-size: $font-size-md;
	}

	.active {
		font-weight: 800;
		text-decoration: underline $color-vidid;
		text-decoration-thickness: 5px;
	}
}
