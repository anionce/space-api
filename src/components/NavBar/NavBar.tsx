import { NavLink } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import './NavBar.scss';
import React from 'react';

export const NavBar = () => {
	const { t } = useTranslation();

	return (
		<nav>
			<NavLink to='capsules'>{t('space_api.navbar.capsules')}</NavLink>
			<NavLink to='rockets'>{t('space_api.navbar.rockets')}</NavLink>
			<NavLink to='rocket/add'>{t('space_api.navbar.add')}</NavLink>
		</nav>
	);
};
