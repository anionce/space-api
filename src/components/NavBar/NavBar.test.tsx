import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { NavBar } from './NavBar';

test('renders correcly', () => {
	const { container } = customRender(<NavBar />);

	expect(container).toMatchSnapshot();
});

test('renders elements correctly', () => {
	customRender(<NavBar />);

	expect(screen.getByText(/Capsules/i)).toBeInTheDocument();
	expect(screen.getByText(/Rockets/i)).toBeInTheDocument();
	expect(screen.getByText(/Add a rocket/i)).toBeInTheDocument();
});
