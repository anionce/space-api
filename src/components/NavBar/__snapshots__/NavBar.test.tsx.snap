// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <nav>
    <a
      class=""
      href="/capsules"
    >
      Capsules
    </a>
    <a
      class=""
      href="/rockets"
    >
      Rockets
    </a>
    <a
      class=""
      href="/rocket/add"
    >
      Add a rocket
    </a>
  </nav>
</div>
`;
